### ISSUE TITLE
[QA] Chrome DevTools Protocol

### ISSUE THREAD

**Issue by kjnanda** [GoogleChromeLabs/ps-analysis-tool#280]
# Scenarios verified for CDP

<google-sheets-html-origin>
Module | Scenario | Status
-- | -- | --
CDP | By default, Use CDP switch will enabled | Pass
  | After installing the extension and opening a new URL Observe two things here:- Ensure that the 'Use CDP' switch is activated.- Confirm the visibility of the debugger notic | Pass
  | Changing the state of Use CDP switch will take some time to reflect the data back in PSAT | Pass
  | Turning off the CDP:- The debugger notice should disappear; please allow some time for this change.- No errors should occur on the chrome://extension page.- A page refresh is necessary to obtain the correct information.- Blocked cookies should no longer be highlighted in yellow. | Pass
  | Turning on CDP:- Debugger notice should reappear (may take a moment).- No errors should occur on the chrome://extension page.- Perform a page refresh for accurate information.- Blocked cookies should be highlighted in yellow. | Pass
  | Table : Cookie Accepted column will not present any moreFilter - Cookie Blocked option should be presentFilter : Cookie Blocked reason with different filters option | Pass
  | With CDP Enabled Cookies - PSAT should show more information with - partitionKey, - Priority, - Size, - Blocked reason in filter | Pass
  | Cookies those are blocked will be visible with Yellow color bg color | Pass
  | Cookies with warning should not have Yellow BG color | Pass
  | Selecting blocked cookie should have diff bg color | Pass
  | Clicking on any blocked cookie it show more info with - Warning- Blocked Reason | Pass
  | Partitioned key: The cookie should have the partition key when that same cookie is set under response header with partitioned attribute | Pass
  | Sorting with Partition Key, Priority and size should work as expected | Pass
  | Cookie preview area should be scrollable | Pass
  | Cookies should be loaded without any delay | Pass
  | The count of total cookies on cookies landing page and pop up should match | Pass
  | Cookies counter reset issue | Pass
  | The order the columns should be as per the Application Tab | Pass
  | THere should not be any flicker effect while cookies are being loaded | Pass
  | Filters should newly added options  - Size - Priority - cookies blocked reason | Pass
  | Should have priority and size value for every single column | Pass
  | Sorting, searching and filtering should work as expected | Pass
  |   |  
Chrome://extensions | There should not be any errors from debugger | Pass
  | Turning on/off CDP switch should not throw errors | Pass
  | Clicking on "Analyse this tab" button from pop up and cookies landing page should not throw any errors | Pass
  | Clearing filter options should not throw any errors | Pass
  | Clicking on any frames from the sidebar should not throw errors | Pass
  |   |  
Issue Triangulation | Frame overlay should shows all the information that we aim to look in the tooltip like Page Frame, No. of blocked cookies, and Blocked reason. | Pass
  | IT should shows the correct count of each cookies type on tooltipFPSc3PCsBlocked 3PCs | Pass
  | Filters are working as expected. Changing the filter, it should reflect on tooltip | Pass
  |   |  
Copy Network string | Right clicking on any row from the table should copy the string | Pass
  | Pasting the same string in Network tab should display the requests | Pass
  |   |  
Cookies Landing page > Cookie Blocked Matrics | THere should not be any UI glitch | Pass
  | Cookies landing page have 3 top sections:1. Cookies Insight2. Blocked Cookies3. Frame | Pass
  | Tooltip should be relevant to the block it refers to | Pass
  | Responsive mode - UI should not break | Pass

--------

### Blocked cookie insight QA

- ~**There should be semi separator as shown in the screenshot in Frames section to make it uniform**~ We won't be adding border bottom.
Please check the stable version's cookies landing page for more clarity.

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/6618fe42-4b7c-42ec-b488-d3be1253177e)</kbd>

-----

- [x] **The top margin does not look good!**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/73d9429c-26a6-49c6-9330-53c1e6917941)</kbd>

-----

- [x] **There seems an issue with total frames count**

`Test URL - https://www.aljazeera.com/`

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/8f6f2a0e-0bd6-4b11-b53e-f545de8fbced)

------

- [x] **We should change the color for better readability.** _Low Priority_

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/acffb9cb-a366-41f8-b551-bb36de250470)</kbd>

----

- [x] **When there is no cookie present UI looks broken. Needs proper padding for the error message.**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/4325b671-6106-49f6-86ff-bbbd7e8bbea0)</kbd>

-----

- [x] **The circle is not present when there is only `1` frame is visible**

E.g. Check on example.com domain to replicate this issue

| Dark Theme | Light Theme |
|--- |--- | 
|![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/d72cde9a-835e-41f7-ae29-dd3e4a656a90) | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/4d448fd4-0f71-4b29-9305-9a4b4c02b634) |

------

- [x] **Cookies landing page > Information overlaps while clicking on expand/collapse button**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/a2732e0c-c0c3-4f07-bcde-b89071a82ad2

-----

- [x] **UI looks broken** 

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/59a9e055-b918-4c1e-b153-1f0932f12288)</kbd>

Question: Do we have fixed the position of the matrix to be shown?

-----

- [x] **Cookie blocked reason - expand view : Missing the desc for some cookies**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/ae5251ac-5ea0-40cc-9da2-81a336cd52b4)</kbd>

------

- [x] **Should have some gap between "Blocked Reason and Warnings"**

`Test URL - https://www.livemint.com/`

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/38eb918a-5e70-4434-8a8e-ad5b3677c1ca)</kbd>

------

- [x] **How these cookies comes under both category? Warning and Blocked both?**

`Test URL - https://www.livemint.com/`

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/20dd446b-ff31-4f34-8efe-300f5c9101d7)<kbd>

------

- [x] **UI glitch in responsive mode**

We have similar UI for Cookie matrix. Can we limit the length the of horizontal line I high lighted in the screenshot below to make it uniform?

| Cookie Matrix UI | Blocked Cookie Matrix UI |
|---- |---- |
| ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e39eecfb-8a69-4e7d-9bb5-675738330751) | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/93858c88-44d6-4937-9833-707be47a6236) |

-----

- [X] **With the latest code pull it gives an error on extension page without performing anything.** I don't see this errors now but there are more new errors. Added [here](https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/280#issuecomment-1851328502) 

<kbd><img width="463" alt="Screenshot 2023-12-11 at 17 13 06" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/fff55231-f83a-4856-961a-4e1ef27fcdd8"></kbd>

-------

- [x] **Issue with total count on cookies landing page**

| Tab | Screenshot|
|---- |----- |
| Popup and cookies landing page | <kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/cbf57000-2c45-46e0-957c-a632df154856)</kbd>| 
| 1st frame - 3PCs count | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/8cb96f7b-0b92-4031-87b4-7ffc1a262780) |
| 2nd frame - 3PCs count | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/638a91b4-7464-4ed8-aff6-8733c3bd9f7c)
| Cookies Landing page shows wrong count for 3PCs | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/ccfaaae4-aeb5-40ee-9b09-6758411a5dcc) |

-------

- [x] **Sorting by size (ascending and descending) order is not working as expected.**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/ac9f092f-3331-46ad-9882-5efc6a54b1cc

------

- [x] **Issue with Partition Key**

| Tab | Meta information | Screenshot/visual |
|---- |---- | ----- |
| Test Site  | https://www.aljazeera.com | - |
| Tested against | feat/cdp branch (latest [commit](096c35412af543faa1b42b4bbad6c5fda512a7f3) | - |
| Cookie name | receive-cookie-deprecation | - |
| Application Tab > Shows partitioned ? | Yes | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/d76aa7fe-cf31-4df4-ac73-d0b2a2360aa6) |
| Network Tab > Response Cookie > Shows partitioned Attribute ? | Yes | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/555ab0f4-11a7-4017-961d-f569d51b3f62)|
| Response Header > Set-cookie > Shows partitioned Attribute ?  | <li>Yes <br/> <li>chrome-default & chrome-3pcd - Shows the same result  | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e56c96a6-4027-4e41-ba84-0ccdb25daf9c)|
| PSAT Extension > Shows partition attribued ? | No | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/9060e7e5-d989-47bb-8d89-89dbaad4527e)|

-----

- [x] **Found this cookie in PSAT as blocked on default chrome instance but it was not present win Network tab!**

<kbd><img width="1280" alt="Screenshot 2023-12-07 at 15 26 01" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c0d55fc2-5cd0-4c23-aebc-fcfbf973e6f0"></kbd>

-----

- [x] **`GreCAPTCHA` should have warning under description.**

| Tab | Screenshot |
|--- |---- |
| Console Tab | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/2bf6d3c2-dbff-4f6a-9d2d-928e19106514) |
| PSAT | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/50bf2bda-e37b-4567-9212-1fded24db1e2) |

-----

- [x] **Cookie Preview area - Please make it scrollable to view the complete details of the selected cookie.**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e74c94de-47dc-4f98-ae76-34c6f09b12e6

------

- [x] **Noticed a delay in showcasing the cookie as blocked.**

Observation: 
- When I visit any frame for the first time, for some frames (especially when the no of cookies are more or needs scrolling to view the cookies) I have noticed a delay in showing those cookies with yellow colored back ground. 

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c16aec4d-6b7f-4789-8285-e7b61919fbed

------

- [x] **There is no clear indication of the row being selected while clicking on yellow colored row.**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/31d6873a-78e2-4677-bdbb-0ee12dcb8c4a

------

- [x] **Cookies set in the response header shows as blocked in PSAT extension**

| Meta information | Details (with screenshot if required) |
|---- |-----|
| Chrome Instance | with default environ (without 3PCD ) |
| Test URL | https://timesofindia.indiatimes.com/ |
| Cookie name | 3pi |
| Network Tab blocks this cookie? | ❎ |
| Application Tab shows this cookie? | ✅ |
| PSAT Extension shows this cookie? | ✅  but as blocked. The reason is the cookies has SameSite=None; Secure and not Partitioned |
| Network Tab | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/97af17be-e8d4-495e-b5ca-496cc6a4020c) |
| Application Tab | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c4d7d062-6a5d-4ad6-b7e5-4185c8b89d47) |
| PSAT Extension | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/4736489e-08de-4382-9727-83fa5e4181ce) |

The reason is correct for blocking but this should be in 3PCD environment. Since the response header shows `3pi` under Set Cookies it should not blocked.

-------

- ~Please remove the extra pixel gap.~ Invalid issue

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/915011ac-5fb1-4321-9b0a-e6bb2460df95)

------

- [X] **Changing the state of `Use CDP` switch does not reflect the change in PSAT extension immediately.** This is the accepted behaviour as mentioned [here](https://rtcamp.slack.com/archives/C0526784HL4/p1702274980375059?thread_ts=1702266889.777989&cid=C0526784HL4).

- It takes noticeable amount of time when doing this on the website where data keeps updating frequently. After the page is refreshed it shows the correct data.
- It take little less time than mentioned above but the change in the state does not effect immediately

Correct data = Cookies with/without yello color based on switch

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/ddd22539-4e21-4334-b1c5-4d821eddcacb

-------

- [x] **Turning on/off the switch throws the error on chrome://extension page.**

<kbd><img width="675" alt="Screenshot 2023-12-01 at 15 38 22" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/f63b18c9-089c-474a-8de7-2775932aedcc"></kbd>

-------

- [X] **Activating or deactivating CDP for a specific tab should be confined to that individual tab. Currently, once enabled, it remains active by default for all other tabs, which is not the intended behavior** Answer : [Comment](https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/280#issuecomment-1837048093) 

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/7d806ee0-acaf-4253-98f4-a4ccd5626463

This issue still persists

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/12daf167-7c3b-462d-b902-3650541fd9b6

------

- [x] **Disabling the CDP does not eliminate the debugger notification**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/f5fa1306-27e1-4b85-8185-a7744f37142e

-----

- [x] **Observed a significant amount of flickering/glitch when using the CDP. Why does it keep loading?**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e9ba0807-e1c1-4982-82be-c971f24b992a

------

- [x] **Debugger does not appear immediately upon reactivating the CDP switch from the popup; the debugger becomes visible only when the 'Analyse this tab' button reappears**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e6cdd81a-aca0-425d-8762-a10a1eb9d232

----

- [x] **Inconsistencies in various parameters when multiple cookies share the same name and value**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/a4fea22a-f07c-46d5-8425-d1fff9b455af)</kbd>

-----

- [x] **"Size" - It is visible under filters but I don't see the column for the same under the table**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/2ce60fa3-6696-4e63-a904-b914b4b8510c)</kbd>

-----

- [x] **Partition Key Column - it continuously flickers and fails to remain stable**

In this visual please observe the `Partition key column` 

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/fdc990be-5f38-4f21-93d6-28e587b33145

----

- [x] **Application Tab shows the priority for almost each rows where as PSAT fails to show priority for each rows. Mostly I have observed the priority is either medium or empty. Could not find any rows with High or low priority**

| Application Tab | PSAT |
|---- |----- |
|![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/00a3d9f5-47b8-4185-adcf-f020e31ed2b1) | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/11d1ea14-51e1-4d4f-8b28-f280065d1f4c) |

-----

-  ~**`Blocked Reason` -  Please check the screenshot below where it shows that cookie is blocked because the `SameSite is Lax` where as `SameSite` column has blank value!**~ - Not Valid - We have removed the column and showing the reason in desc area only.

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/9bf9c476-32fd-4c33-9111-c28afd72b31f)</kbd>

------

- [x] **`Expire/Max-Age` - Shows incorrect data**

`Test URL - https://www.bbc.com/news`

| Test Environment | Screenshot |
|------ |------ |
| **Latest stable PSAT version - V0.3.2** | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/3ce33ff3-aaab-407f-9eda-0698b1cb71b0) |
| **CDP - Chrome default env** | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/5a73e013-2e22-433f-8724-a6fae768eaa5)
| **CDP - 3PCD Chrome env** | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/da76ab19-b775-45e0-ab21-5e5105bb3890) |

**With the latest code pull - It is not fixed** 

| Tab | Screenshot |
|------ |------ |
| **Network Tab** | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/95f4a474-f3bb-429c-8ca0-b2188482d218) |
| **Application Tab** | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c89ac65e-b425-4f11-bd64-ce16df57974d)|
| **CDP - 3PCD Chrome env** | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/ba90e844-2960-4465-a600-bbfe5beeb135) |

-------

**Comment by kjnanda** [comment]
@amovar18 Once the flicker issue is resolved, I'll be able to thoroughly test sorting and filtering functionalities.

**Comment by amovar18** [comment]
>Activating or deactivating CDP for a specific tab should be confined to that individual tab. Currently, once enabled, it remains active by default for all other tabs, which is not the intended behavior

This is not in our control because CDP will show the notification for the whole browser.

**Comment by amovar18** [comment]
>  Turning on/off the switch throws the error on chrome://extension page.

Not able to reproduce this.

> "Size" - It is visible under filters but I don't see the column for the same under the table

Fixed this already in latest code.

**Comment by amovar18** [comment]
> Found this cookie in PSAT as blocked on default chrome instance but it was not present win Network tab!

Please keep the network tab open as soon as you load the timesofindia.indiatimes site since its request is not captured if devtools is not opened.

**Comment by amovar18** [comment]
> Changing the state of the `Use CDP` switch does not reflect the change in the PSAT extension immediately 

This is expected as the process takes some time to detach the debugger from the tab. This is because the process is async in nature.

**Comment by kjnanda** [comment]
@amovar18 Please check my comments below

--------

- [ ]   Issue with Partition Key

This issue has not resolved completely. THere are two cookies with the same name but for one it has discrepancy in showing the result wrt to Application and network tab. 

- Exp/Max-Age has different value than Application Tab.
- Size, priority, HTTP columns does not show the values

| Application tab | PSAT |
|--- |---- |
| ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/1d4df2ed-5eeb-4124-896b-501d80436637) | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/22a3453e-7696-45e4-ac91-07e60288657a) |

-----

- [ ] Clicking on 'Clear All' button throws the error

<img width="689" alt="Screenshot 2023-12-11 at 12 02 35" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/efcb59d1-f2cd-4320-945e-0f1974935672">

**Comment by kjnanda** [comment]
@amovar18 Retested below fix but it has not been fixed completely.

> Issue with Partition Key

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/92263ea8-a8d8-4df1-a48c-47509a89b707)</kbd>

Also, these all are coming from 3rd party domain then why only one cookie is getting blocked? 
<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/d2c0a7a2-71e0-4fdf-a37f-a2af8c8065da)</kbd>

**Comment by amovar18** [comment]
> How these cookies comes under both category? Warning and Blocked both?

This is possible because reasons are different blocked was because blocked reason was url was not as same as the domain, and warning is to prepare for 3pcd

**Comment by kjnanda** [comment]
Issues that we can add in next release : 

- [ ] **An issue with turning off/on the CDP switch** Can be the part of next release  Ref: [comment](https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/280#issuecomment-1851881548) 

`Steps:`

1. Open any URL in the browser with fresh chrome profile 
2. Please make sure to turn off the CDP switch 
3. Click on `Analyze this tab` button
4. Observe the cookies being fetched
5. These columns : `Partitioned Key, Priority and Size` will remain empty which is expected behaviour. ✅ 
6. Open the filter from any one frame and observe the `Cookies Blocked` option which shows only one value as **False** Is this expected? ❓ 
7. Now, turning on the CDP switch from Pop up and observe 
         - The `Cookies Blocked` option has true and false both values
         - The blocked cookies turned into Yellow 
         - These columns : `Partitioned Key, Priority and Size` will remain empty 

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/b4bd6cde-ed48-495a-b7f9-2328888ed7d7)</kbd>

**Comment by kjnanda** [comment]
@amovar18 Please check my comment below.

- [ ] **Why this fonts are too small?**

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/12cd9a15-63f6-4683-a896-5c4088b22c1c)

----

- [ ] **Code formatting is partially fixed. Please use single or double quote if it is not possible to omit "`"**

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/2f6fbda2-21b1-4b06-96c9-b5b1b45e4163)

**Comment by amovar18** [comment]
>  Why this fonts are too small?

Its because the first line is the heading the second line is the description of that heading.

> I don't see any changes

I haven't changed the line height because the place you highlighted was for 2 different reasons.

**Comment by kjnanda** [comment]
These fonts are really small that create trouble in reading. Same applies on cookies landing page

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/0d3607a0-b3df-4bb4-8f34-04a60bcd32b1)

----

>  haven't changed the line height because the place you highlighted was for 2 different reasons.

BUt since you marked it fixed I checked it.

**Comment by kjnanda** [comment]
@amovar18 Some feedback on fonts for warning or blocked cookie

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/841d6ef7-1bb7-4f97-b87f-817ec4be51ad)</kdb>

**Comment by amovar18** [comment]
> There seems an issue with total frames count

Outermost frame + fenced frame + subframe = total frame.
Outermost frame refers to the top frame. The page URL is top frame.

>  The circle is not present when there is only 1 frame is visible

Its because there is no metric to add the categorise the frame in.

**Comment by kjnanda** [comment]
@amovar18 

> Outermost frame + fenced frame + subframe = total frame.
Outermost frame refers to the top frame. The page URL is top frame.

Then should not it be named as "Top frame or main frame" instead of outer frame?

**Comment by kjnanda** [comment]
After the latest code pull from `feat/cdp` branch there are some issues that need to be fixed.

- [ ] **Flicker effect in siderbar with error on chrome://extension page**

Steps:

1. Open any URL and cookies will be loaded
2. Open a new tab > Enter URL > Click on "Analyze this tab" button
3. Observe the errors thrown on chrome://extensions page 
4. Goto Tab 1 and Click on "Analyze this tab" button again
5. Observe the flickering in the sidebar


https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/2ddf71e7-c323-468c-8a06-5446df95a4d8

<kbd><img width="1470" alt="Screenshot 2023-12-15 at 18 02 55" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/7b6717ef-932c-4778-86de-e13c0e81d898"></kbd>

**Comment by kjnanda** [comment]
Closing this issue as it is addressed in V0.4.0 release
